src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,62 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import java.io.InputStream
+
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+    val buf1 = ByteArray(4096)
+    val buf2 = ByteArray(4096)
+
+    var size = 0L
+    while(true) {
+        val len1 = actual.read(buf1)
+        val len2 = other.read(buf2)
+
+        when {
+            len1<len2 -> expected("to have the same size, but actual has size ${size+len1} which is smaller then the expected stream")
+            len1>len2 -> expected("to have the same size, but the expected stream size ${size+len2} which is larger then the acual stream")
+            len1==-1 -> return
+        }
+
+        for(i in 0..len2) {
+            val b1 = buf1[i]
+            val b2 = buf2[i]
+            if(b1!=b2) {
+                expected("to have the same content, but found first difference at pos ${size+i}")
```

Comment:
Should report what the difference is as well as it's position.

Reply:
Imagine a file with a lots of differences. I'm not sure that the information what is different is really useful. I could ofc show the values of the first different byte.

Reply:
That's what I was thinking, just the first difference, isn't that what it's doing currently?